### PR TITLE
Add type hints to shapes

### PR DESCRIPTION
**Pull request by gran4**

### CODE REVIEW

File: pyglet/shapes.py [hunk 1]
```diff
@@ -42,18 +42,28 @@ def on_draw():
...
 from pyglet.gl import glBlendFunc, glEnable, glDisable
 from pyglet.graphics import Batch, Group
+from pyglet.graphics.shader import ShaderProgram, ComputeShaderProgram
+
+from typing import (
+    Union,
+    Optional,
+    Tuple,
+    List
+)
```
Comment (shenjackyuanjie) [inline pyglet/shapes.py:63]: how about just `from typing import Union, Optional, Tuple, List`

File: pyglet/shapes.py [hunk 2]
```diff
@@ -116,7 +126,7 @@ class _ShapeGroup(Group):
     sharing the same parent group and blend parameters.
     """
 
-    def __init__(self, blend_src, blend_dest, program, parent=None):
+    def __init__(self, blend_src: int, blend_dest: int, program: ShaderProgram, parent: Optional[Group]=None) -> None:
```
Comment (shenjackyuanjie) [inline pyglet/shapes.py:129]: why not add a gap between `Optional[Group]` with `=None`
like `parent: Optional[Group] = None`

File: pyglet/shapes.py [hunk 3]
```diff
@@ -431,16 +449,18 @@ def batch(self, batch):
 class Arc(ShapeBase):
     _draw_mode = GL_LINES
 
-    def __init__(self, x, y, radius, segments=None, angle=math.tau, start_angle=0,
-                 closed=False, color=(255, 255, 255, 255), batch=None, group=None):
+    def __init__(self, x: number, y: number, radius: float, segments: Optional[int]=None, 
+                 angle: float=math.tau, start_angle: float=0, closed: bool=False, 
+                 color: Tuple[int, int, int, int]=(255, 255, 255, 255), 
+                 batch: Optional[Batch]=None, group: Optional[Group]=None) -> None:
```
Comment (shenjackyuanjie) [inline pyglet/shapes.py:455]: add gap between type hint and default value

File: pyglet/shapes.py [hunk 4]
```diff
@@ -1395,7 +1424,7 @@ def border_color(self, values):
         self._update_color()
 
     @property
-    def color(self):
+    def color(self) -> Tuple[int, int, int, int]:
```
Comment (shenjackyuanjie) [inline pyglet/shapes.py:1427]: how about add a type named `color rgba`

**Review by shenjackyuanjie — COMMENTED** [review]
add some gap?

File: pyglet/shapes.py (hunk 1 above)
Comment (gran4) [reply pyglet/shapes.py:63]: As imports get longer, it is better to put them on separate line. I do not actually know how many are needed to start putting them on separate lines though.

File: pyglet/shapes.py (hunk 2 above)
Comment (gran4) [reply pyglet/shapes.py:129]: Just makes the line longer and does not add anything

File: pyglet/shapes.py (hunk 4 above)
Comment (gran4) [reply pyglet/shapes.py:1427]: Maybe

**Comment by gran4** [comment]
In the ellipse class, segments is not in the docstrings. Is this a bug @benmoran56 or is this something that just needs to be updated?

File: pyglet/shapes.py (hunk 3 above)
Comment (gran4) [reply pyglet/shapes.py:486]: I think that will be a waste of time. It will make the lines longer for no reason.

**Pull request closed without merging**